pkg/prometheus/operator.go

```diff
@@ -229,6 +229,16 @@ func (sr *StatusReporter) Process(ctx context.Context, p monitoringv1.Prometheus
 		}
 	}
 
+	// Indicate that no scrape resource selectors are specified. Do not take precedence over an unhealthy condition.
+	allSelectorsNil := commonFields.ServiceMonitorSelector == nil &&
+		commonFields.PodMonitorSelector == nil &&
+		commonFields.ProbeSelector == nil &&
+		commonFields.ScrapeConfigSelector == nil
+	gotCondition := sr.Reconciliations.GetCondition(key, p.GetObjectMeta().GetGeneration())
```

Comment:
reading the code, I realize that we could rename `GetCondition`() to `GetReconciledCondition`() to facilitate the understanding.
Also `gotCondition` -> `reconciledCondition`.